### ISSUE TITLE
The "battery" option in Energy Saver

### ISSUE THREAD

**Issue by jingquanquan2018** [osy86/HaC-Mini#107]
OC v2.3.1
![image](https://user-images.githubusercontent.com/44221069/68067838-bb855300-fd87-11e9-8e8d-d867458de1b1.png)
I think it shoushouldn't be visible on mac mini

**Comment by osy86** [comment]
Okay, I spent way too much time on this issue.

That text shows up when `IOPSPowerSourceSupported` returns true for `Battery Power`. That function returns true if `system-type` in `IOService:/` registry entry is equal to `02`. You'll notice that the NUC has it set to `02`. Why is that? So that entry comes from the ACPI table. The FADT defines the power management profile. 02 means "mobile".

So to "fix" it, we can either fake the FADT to not return 02 (which might have other side effects) or to patch system preferences to not show that string (which is easy but useless).

Because this doesn't cause any issues and everywhere else, there's no report of a battery present, there's no point fixing it.

**Comment by jingquanquan2018** [comment]
Sure，it doesn't matter

**Comment by chrisallen** [comment]
@osy86 I'd be happy to test a patch that fakes the FADT to not return 02 if you like.

**Comment by osy86** [comment]
Sure

**Comment by osy86** [comment]
Please also get the output of `ioreg -p IOPower -l` before and after.

**Comment by chrisallen** [comment]
Here is the "Before" https://gist.github.com/chrisallen/96af391e74bef30181e844f835152562

**Comment by osy86** [comment]
Try this in config.plist
```xml
			<dict>
				<key>Comment</key>
				<string>patch FADT from mobile profile to desktop</string>
				<key>Enabled</key>
				<true/>
				<key>Find</key>
				<data>
				AAIJALIAAACgoQAAABgAAA==
				</data>
				<key>Replace</key>
				<data>
				AAEJALIAAACgoQAAABgAAA==
				</data>
			</dict>
```

Also run `ioreg -l | grep system-type` and make sure it's changed to `01`.

**Comment by chrisallen** [comment]
```
❯ ioreg -l | grep system-type
    |   "system-type" = <01>

~
```

![image](https://user-images.githubusercontent.com/203867/74988780-a998fe80-54a3-11ea-8a3f-488ba82e737c.png)


Here is the "After" https://gist.github.com/chrisallen/7319edd621c90ad3115756a3ae536ea2

**Comment by osy86** [comment]
I don't see any difference in the hibernation option, but why don't you keep it like this and let me know (in the other issue) if you still get unexpected shutdowns.

**Comment by chrisallen** [comment]
Yeah, I'm happy to keep it like this. One thing I have noticed (and not sure if it is new) is that as soon as I manually sleep the NUC, I see the power led "breathe" for a moment then the system wakes back up.  I have disabled "wake for ethernet network access" but perhaps its the "power nap" feature.

I also noticed that I have not seen a USB device disconnected message when it wakes in this scenario (I have a USB3 SSD drive connected to the rear blue USB ports)

Is there any logs or data I can grab for you?
